scripts/create_sfm_from_obj.py

```diff
@@ -0,0 +1,145 @@
+                                         write_cameras_binary)
+
+
+def render_dataset_and_get_colmap_images_and_cameras(mesh_path, dataset_path, 
+                                                     fx, fy, cx, cy, W, H):
+    mesh_path = args.mesh_path
+    Rs, Ts, mesh = create_look_at_poses_for_mesh(mesh_path, subdivisions=2)
+    dataset_path.mkdir(parents=True, exist_ok=True)
+    mapping_path = dataset_path / 'mapping'
+    mapping_path.mkdir(parents=True, exist_ok=True)
+    colmap_images = {}
+    colmap_camera_id = 1
+    colmap_camera = create_colmap_camera(W, H, fx, cx, cy, k1=0.)
+    colmap_cameras = {colmap_camera_id: colmap_camera}
+    for i in tqdm(range(Rs.shape[0])):
+        R = Rs[i].unsqueeze(0)
+        T = Ts[i].unsqueeze(0)
+        image = render_image(mesh, fx, fy, cx, cy, W, H, R, T)
+        image = (image[..., :3] * 255.).astype(np.uint8)
+        image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)
+        img_name = ('%d.png' % (i + 1)).zfill(8)
+        cv2.imwrite(str(mapping_path / img_name), image)
+
+        colmap_image_name = 'mapping/' + img_name
+        colmap_image_id = i + 1
+        colmap_image = create_colmap_image_from_pytorch3d_RT(R[0], T[0], 
+                                                          colmap_image_name,
+                                                          colmap_image_id,
+                                                          colmap_camera_id)
+        colmap_images[colmap_image_id] = colmap_image
+    return colmap_images, colmap_cameras
+
+def create_sfm_from_colmap_images_and_cameras(dataset_path, output_path,
+                                              colmap_images, colmap_cameras): 
+    # Configure SFM
+    images = Path(dataset_path)
+    outputs = Path(output_path)
+    sfm_pairs = outputs / 'pairs-sfm.txt'
+    loc_pairs = outputs / 'pairs-loc.txt'
+    features = outputs / 'features.h5'
+    matches = outputs / 'matches.h5'
+    raw_dir = outputs / "raw"
+    sfm_dir = outputs / "sfm"
+    feature_conf = extract_features.confs['superpoint_max']
+    matcher_conf = match_features.confs['superglue']
+    matcher_conf['model']['weights'] = 'indoor'
+    references = [str(p.relative_to(images)) for p in (images / 'mapping/').iterdir()]
+    print(len(references), "mapping images")
+    
+    # Extract features and matches
+    extract_features.main(feature_conf, images, image_list=references, feature_path=features)
+    pairs_from_exhaustive.main(sfm_pairs, image_list=references)
+    match_features.main(matcher_conf, sfm_pairs, features=features, matches=matches)
+    pairs = sfm_pairs
+    assert features.exists(), features
+    assert pairs.exists(), pairs
+    assert matches.exists(), matches
+
+    # Create a database and import images, features and matches
+    raw_dir.mkdir(parents=True, exist_ok=True)
+    database = raw_dir / 'database.db'
+    create_empty_db(database)
+    camera_mode = 'AUTO'
```

Comment:
Pull out as an arg to the func. Also what does this mean?

Reply:
These are camera modes in pycolmap. I changed this to using the enum name, rather than hardcoding the string.